File: postman/sdks/Collections SDKs/iconocracy-corpus/java/src/main/java/com/iconocracycorpussdk/services/PostService.java
package com.iconocracycorpussdk.services;

import com.fasterxml.jackson.core.type.TypeReference;
import com.iconocracycorpussdk.config.IconocracyCorpusSdkConfig;
import com.iconocracycorpussdk.config.RequestConfig;
import com.iconocracycorpussdk.exceptions.ApiError;
import com.iconocracycorpussdk.http.Environment;
import com.iconocracycorpussdk.http.HttpMethod;
import com.iconocracycorpussdk.http.ModelConverter;
import com.iconocracycorpussdk.http.util.RequestBuilder;
import com.iconocracycorpussdk.models.PostDataRequest;
import java.util.Optional;
import java.util.concurrent.CompletableFuture;
import lombok.NonNull;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;

/**
 * PostService Service
 */
public class PostService extends BaseService {

  private RequestConfig postDataConfig = RequestConfig.builder()
    .environment(Environment.POSTMAN_ECHO)
    .build();

  /**
   * Constructs a new instance of PostService.
   *
   * @param httpClient The HTTP client to use for requests
   * @param config The SDK configuration
   */
  public PostService(@NonNull OkHttpClient httpClient, IconocracyCorpusSdkConfig config) {
    super(httpClient, config);
  }

  /**
   * Sets method-level configuration for {@code postData}.
   * Method-level overrides take precedence over service-level configuration but are
   * overridden by request-level configurations.
   *
   * @param config The configuration overrides to apply at the method level
   * @return This service instance for method chaining
   */
  public PostService setPostDataConfig(RequestConfig config) {
    this.postDataConfig = config;
    return this;
  }

  /**
   * This is a POST request, submitting data to an API via the request body. This request submits JSON data, and the data is reflected in the response.
   *
   * A successful POST request typically returns a `200 OK` or `201 Created` response code.
   *
   * @param postDataRequest {@link PostDataRequest} Request Body
   * @return response of {@code Object}
   */
  public Object postData(@NonNull PostDataRequest postDataRequest) throws ApiError {
    return this.postData(postDataRequest, null);
  }

  /**
   * This is a POST request, submitting data to an API via the request body. This request submits JSON data, and the data is reflected in the response.
   *
   * A successful POST request typically returns a `200 OK` or `201 Created` response code.
   *
   * @param postDataRequest {@link PostDataRequest} Request Body
   * @return response of {@code Object}
   */
  public Object postData(@NonNull PostDataRequest postDataRequest, RequestConfig requestConfig)
    throws ApiError {
    RequestConfig resolvedConfig = this.getResolvedConfig(this.postDataConfig, requestConfig);
    Request request = this.buildPostDataRequest(postDataRequest, resolvedConfig);
    Response response = this.execute(request, resolvedConfig);
    byte[] bodyBytes = ModelConverter.readBytes(response);
    return ModelConverter.convert(bodyBytes, new TypeReference<Object>() {});
  }

  /**
   * This is a POST request, submitting data to an API via the request body. This request submits JSON data, and the data is reflected in the response.
   *
   * A successful POST request typically returns a `200 OK` or `201 Created` response code.
   *
   * @param postDataRequest {@link PostDataRequest} Request Body
   * @return response of {@code CompletableFuture<Object>}
   */
  public CompletableFuture<Object> postDataAsync(@NonNull PostDataRequest postDataRequest)
    throws ApiError {
    return this.postDataAsync(postDataRequest, null);
  }

  /**
   * This is a POST request, submitting data to an API via the request body. This request submits JSON data, and the data is reflected in the response.
   *
   * A successful POST request typically returns a `200 OK` or `201 Created` response code.
   *
   * @param postDataRequest {@link PostDataRequest} Request Body
   * @return response of {@code CompletableFuture<Object>}
   */
  public CompletableFuture<Object> postDataAsync(
    @NonNull PostDataRequest postDataRequest,
    RequestConfig requestConfig
  ) throws ApiError {
    RequestConfig resolvedConfig = this.getResolvedConfig(this.postDataConfig, requestConfig);
    Request request = this.buildPostDataRequest(postDataRequest, resolvedConfig);
    CompletableFuture<Response> futureResponse = this.executeAsync(request, resolvedConfig);
    return futureResponse.thenApplyAsync(response -> {
      byte[] bodyBytes = ModelConverter.readBytes(response);
      return ModelConverter.convert(bodyBytes, new TypeReference<Object>() {});
    });
  }

  private Request buildPostDataRequest(
    @NonNull PostDataRequest postDataRequest,
    RequestConfig resolvedConfig
  ) {
    return new RequestBuilder(
      HttpMethod.POST,
      resolveBaseUrl(resolvedConfig, Environment.POSTMAN_ECHO),
      "post"
    )
      .setJsonContent(postDataRequest)
      .build();
  }
}

File: postman/sdks/Collections SDKs/iconocracy-corpus/typescript/src/services/post/post-service.ts
import { z } from 'zod';
import { BaseService } from '../base-service';
import { ContentType, HttpResponse, SdkConfig } from '../../http/types';
import { RequestBuilder } from '../../http/transport/request-builder';
import { SerializationStyle } from '../../http/serialization/base-serializer';
import { ThrowableError } from '../../http/errors/throwable-error';
import { Environment } from '../../http/environment';
import { PostDataRequest, postDataRequestRequest } from './models/post-data-request';

/**
 * Service class for PostService operations.
 * Provides methods to interact with PostService-related API endpoints.
 * All methods return promises and handle request/response serialization automatically.
 */
export class PostService extends BaseService {
  protected postDataConfig: Partial<SdkConfig> = { environment: Environment.POSTMAN_ECHO };

  /**
   * Sets method-level configuration for postData.
   * @param config - Partial configuration to override service-level defaults
   * @returns This service instance for method chaining
   */
  setPostDataConfig(config: Partial<SdkConfig>): this {
    this.postDataConfig = config;
    return this;
  }

  /**
 * This is a POST request, submitting data to an API via the request body. This request submits JSON data, and the data is reflected in the response.
A successful POST request typically returns a `200 OK` or `201 Created` response code.
 * @param {Partial<SdkConfig>} [requestConfig] - The request configuration for retry and validation.
 * @returns {Promise<HttpResponse<any>>} - OK
 */
  async postData(body: PostDataRequest, requestConfig?: Partial<SdkConfig>): Promise<any> {
    const resolvedConfig = this.getResolvedConfig(this.postDataConfig, requestConfig);
    const request = new RequestBuilder()
      .setConfig(resolvedConfig)
      .setBaseUrl(resolvedConfig)
      .setMethod('POST')
      .setPath('/post')
      .setRequestSchema(postDataRequestRequest)
      .setRequestContentType(ContentType.Json)
      .addResponse({
        schema: z.any(),
        contentType: ContentType.Json,
        status: 200,
      })
      .addHeaderParam({ key: 'Content-Type', value: 'application/json' })
      .addBody(body)
      .build();
    return this.client.callDirect<any>(request);
  }
}
